packages/compiler/core/checker.ts

```diff
@@ -1755,7 +1754,10 @@ export function createChecker(program: Program): Checker {
   }
 
   function reportAmbiguousIdentifier(node: IdentifierNode, symbols: Sym[]) {
-    const duplicateNames = symbols.map(getFullyQualifiedSymbolName).join(", ");
+    const duplicateNames = symbols
+      .map(getFullyQualifiedSymbolName)
+      .map((name) => (name.includes(".") ? name : `global.${name}`))
```

Comment:
I would check before mapping to the fully qualified name if the symbol has no parent to check its in the global scope instead of checking if it was combined with `.`, Could even be a flag on the getFullyQualifiedSymbolName function to prepend `global.`